pkg/cmd/openshift-sdn/healthcheck-shim.go

```diff
@@ -0,0 +1,55 @@
+package openshift_sdn
+
+import (
+	"sync"
+
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/proxy/healthcheck"
+)
+
+// NotifyingHealthzServer is a shim around the existing proxy healthcheck
+// code, so we can wait until the proxy has successfully reached health.
+// It would be nice to get rid of this in favor of proper status reporting
+// upstream.
```

Comment:
It seems strange to tie into the health-check mechanism but then not actually work via the health-check mechanism. Like, it seems like we ought to just be able to add a `readinessProbe`, and then the SDN pod won't be ready until kube-proxy's health check passes, and we don't need to do anything else. Except that won't really work because there's no correlation between network-plugin-pod readiness and node-readiness-due-to-CNI. But maybe we should try to move toward that somehow.
(Not especially a comment on this PR, just a related thought.)

Reply:
Right. We could solve this with taints, if there was some way to set a taint when the proxy isn't running. That seems like a tricky lifecycle to get.